### PR TITLE
Simplify library code

### PR DESCRIPTION
**Pull request by chanseokoh**
I was looking at the code in .libraries and thought I could improve readability a bit, mainly by using `List` in place of primitive Java arrays.

@akerekes what do you think?

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/LibraryClasspathContainer.java [hunk 1]
```diff
@@ -66,6 +67,6 @@ public String getDescription() {
 
   @Override
   public IClasspathEntry[] getClasspathEntries() {
-    return classpathEntries;
+    return classpathEntries.toArray(new IClasspathEntry[0]);
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/LibraryClasspathContainer.java]: I think this is why the arrays are used. A lot of the Eclipse API around this uses lists.

File: plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/SourceAttacherJob.java [hunk 2]
```diff
@@ -54,37 +56,36 @@ protected IStatus run(IProgressMonitor monitor) {
...
       }
+
+      LibraryClasspathContainer newContainer =
+          libraryClasspathContainer.copyWithNewEntries(newClasspathEntries);
+      JavaCore.setClasspathContainer(containerPath, new IJavaProject[]{ javaProject },
+                                     new IClasspathContainer[]{ newContainer }, monitor);
+      serializer.saveContainer(javaProject, newContainer);
+      return Status.OK_STATUS;
+
     } catch (Exception ex) {
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/SourceAttacherJob.java]: This looks like something we should fix if possible. (not in this PR) Why are we catching java.lang.Exception here?

File: plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/LibraryClasspathContainer.java (hunk 1 above)
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/LibraryClasspathContainer.java]: Yeah, we need extra conversion here, but overall, looks like using a `List` is a win.

File: plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/SourceAttacherJob.java [hunk 3]
```diff
@@ -54,37 +56,36 @@ protected IStatus run(IProgressMonitor monitor) {
     try {
       IClasspathContainer container = JavaCore.getClasspathContainer(containerPath, javaProject);
       if (container instanceof LibraryClasspathContainer) {
```
Comment (akerekes) [inline plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/SourceAttacherJob.java]: Shouldn't the condition be negated? If so, and tests are still passing can you add one that fails on this before fixing it?

Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/SourceAttacherJob.java]: Wow, nice catch!

File: plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/SourceAttacherJob.java (hunk 2 above)
Comment (akerekes) [inline plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/SourceAttacherJob.java]: `sourceArtifactPathProvider` is a `Callable` and this `call()` declares Exception to be thrown. We can either replace it with a `com.google.common.base.Supplier, swallow any exception that happens inside and return a `null` or create an interface similar to `Callable` and declare `CoreException` to be thrown.

**Pull request merged**